trie/goroutinesManager.go

```diff
@@ -0,0 +1,95 @@
+package trie
+
+import (
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-go/common"
+)
+
+type goroutinesManager struct {
+	throttler    core.Throttler
+	errorChannel common.BufferedErrChan
+	chanClose    chan struct{}
+
+	mutex      sync.RWMutex
+	canProcess bool
+}
+
+// NewGoroutinesManager creates a new GoroutinesManager
+func NewGoroutinesManager(
+	throttler core.Throttler,
+	errorChannel common.BufferedErrChan,
+	chanClose chan struct{},
+) (common.TrieGoroutinesManager, error) {
+	if check.IfNil(throttler) {
+		return nil, ErrNilThrottler
+	}
+	if check.IfNil(errorChannel) {
+		return nil, ErrNilBufferedErrChan
+	}
+	if chanClose == nil {
+		return nil, ErrNilChanClose
+	}
+
+	return &goroutinesManager{
+		throttler:    throttler,
+		errorChannel: errorChannel,
+		chanClose:    chanClose,
+		canProcess:   true,
+	}, nil
+}
+
+// ShouldContinueProcessing checks if the processing should be interrupted
+func (gm *goroutinesManager) ShouldContinueProcessing() bool {
+	select {
+	case <-gm.chanClose:
+		return false
+	default:
+		gm.mutex.RLock()
+		defer gm.mutex.RUnlock()
+
+		return gm.canProcess
+	}
+}
+
+// CanStartGoRoutine checks if a new goroutine can be started. If yes, it adds a new goroutine to the throttler
+func (gm *goroutinesManager) CanStartGoRoutine() bool {
+	gm.mutex.Lock()
+	defer gm.mutex.Unlock()
+
```

Comment:
do we need mutex here?

Reply:
Removed it. It will be needed, but I will add it in a future PR in which I will also add tests to prove it works as intended.

Reply:
Yes, because the same manager will be passed on multiple goroutines.